src/app/core/auth/auth.service.ts

```diff
@@ -297,14 +308,52 @@ export class AuthService {
    */
   public getToken(): AuthTokenInfo {
     let token: AuthTokenInfo;
-    this.store.pipe(select(getAuthenticationToken))
+    this.store.pipe(take(1), select(getAuthenticationToken))
       .subscribe((authTokenInfo: AuthTokenInfo) => {
         // Retrieve authentication token info and check if is valid
         token = authTokenInfo || null;
       });
     return token;
   }
 
+  /**
+   * Method that checks when the session token from store expires and refreshes it when needed
+   */
+  public trackTokenExpiration(): void {
```

Comment:
It looks, to me, like we are missing specs for all the new methods (this method, setIdle and isUserIdle) added to this `auth.service.ts`.  Would these be possible to add, or are they overly complex for some reason?